src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4419,7 +4427,8 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
                         parameterType,
                         parameterAnnotations,
                         results[i],
-                        notNullParametersBuilder);
+                        notNullParametersBuilder,
+                        (compareExchangeInfo is object && i == 0) ? compareExchangeInfo : null);
```

Comment:
:memo: I didn't want to add 3 or 4 parameters to the `VisitArgumentOutboundAssignmentsAndPostConditions`, so I've bundled the necessary information into a `CompareExchangeInfo` instance.